### PR TITLE
Keycloak 19 fix realm name

### LINKED ISSUE #143: Metrics contain realm uuid instance of name with KC19
## Description
Adding the metrics-spi to KC19 creates metrics with a UUID value for the 'real' label.

## Expected Behavior
The 'real' label should contain the realm name.

## Actual Behavior
The 'real' label contains a UUID.

## Environment

* Project Versions: 
  * keycloak-metrics-spi : 2.5.3
  * keycloak: 19.0.1

## Steps to reproduce
- Create custom keycloak 19 image with keycloak-metrics-spi jar
- Run container
- Enable metrics-spi Events listener
- Login to Keycloak
- View metrics : curl http://<url>/auth/realms/master/metrics

### PR DESCRIPTION
**Pull request by diversit**
## Motivation
Fixed #143 
In Keycloak 19 the 'realm' in the metrics contain a UUID instead of the Realm name.
Apparently Keycloak changed the value of `event.getRealmId()` from a Realm name to a real Realm id. 

## What
This change will fix this by looking up the Realm by Id via a RealmProvider so the metrics will still contain a Realm name.

Upgraded Keycloak dependency to 19.0.1 to also verify this library is not using any deprecated sources.
Removed the `MetricsEventListenerSpi` since this seems to be an old way of adding a listener and is not needed anymore for v19 Quarkus Keycloak.

## Why
To have readable realm names in the metrics and on the Keycloak dashboard.

## How
Realm name is looked up by using a RealmProvider which is accessible from the KeycloakSession when the MetricsEventListenerFactory is created.

## Verification Steps
1. Build the package, include it in a custom Keycloak 19+ image and start the container.
2. Login into the admin console
3. Enable metrics-listener for events in the Realm settings.
4. Logout
5. Log back into admin console
6. Run `curl -vk https://<keycloak-url>:<port>/auth/realms/master/metrics | grep -v '^#' | grep login` and check metrics output.
The metrics should still contain the realm name like
```
keycloak_logins{realm="admin",provider="keycloak",client_id="account-console",} 1.0
```

## Checklist:

- [x] Code has been tested locally by PR requester
- [x] Changes have been successfully verified by another team member 

## Progress

- [x] Finished task
- [ ] TODO

### CODE REVIEW

**Comment by pb82** [comment]
thanks @diversit 
Does that mean that at the moment using Keycloak 19, you would see UUIDs in the metric labels instead of a realm name?

**Review by pb82 — APPROVED** [review]
LGTM, anyone had a chance to try this yet on KC19?

**Comment by diversit** [comment]
@pb82 Yes, with the current 2.5.3 version on KC 19 the metrics label will contain a UUID for the realm instead of the name. This change fixes this.

**Comment by pb82** [comment]
@diversit is this backward compatible with older Keycloak versions?

File: src/main/java/org/jboss/aerogear/keycloak/metrics/PrometheusExporter.java
Comment (pb82) [inline src/main/java/org/jboss/aerogear/keycloak/metrics/PrometheusExporter.java:499]: Is the realm provider available on older Keycloak versions as well? Should we at least return the `realmId` if we're not able to locate the name here?

**Comment by sonOfRa** [comment]
We're currently in the process of migrating our Keycloak instance to 19 (however still using the Wildfly distribution). I'll apply this patchset without the SPI-file removals and see how they work for us. Since support for Wildfly will get dropped in version 20, rather than version 19, maybe it's best to hold off on removing those files for now?

**Comment by sonOfRa** [comment]
So far, I haven't noticed any problems in our test bed environment with this patch set applied.

**Comment by diversit** [comment]
Can this be merged?

**Comment by pb82** [comment]
@diversit yes, I'd like to verify this against KC 21, once we got the compatibility PR merged. Afterwards, this can be merged too.

**Comment by pb82** [comment]
@diversit I've rebased your changes and merged them in #159 
Thanks!

**Pull request closed without merging**
